src/js/service/SavedStatusService.js

```diff
@@ -1,45 +1,37 @@
 (function () {
   var ns = $.namespace('pskl.service');
 
-  ns.SavedStatusService = function (piskelController) {
+  ns.SavedStatusService = function (piskelController, historyService) {
     this.piskelController = piskelController;
+    this.historyService = historyService;
+    this.lastSavedStateIndex = '';
   };
 
   ns.SavedStatusService.prototype.init = function () {
     $.subscribe(Events.TOOL_RELEASED, this.onToolReleased.bind(this));
     $.subscribe(Events.PISKEL_RESET, this.onPiskelReset.bind(this));
     $.subscribe(Events.PISKEL_SAVED, this.onPiskelSaved.bind(this));
+    this.lastSavedStateIndex = this.historyService.getCurrentStateIndex();
   };
 
-  ns.SavedStatusService.prototype.onPiskelReset = function () {
-    var piskel = this.piskelController.getPiskel();
-    // A first PISKEL_RESET is triggered during the load of a new Piskel, it should be ignored
-    // putting a firstResetDone flag as a nasty workaround for this
-    if (piskel.firstResetDone_) {
-      this.updateDirtyStatus(true);
-    } else {
-      piskel.firstResetDone_ = true;
-    }
+  ns.SavedStatusService.prototype.onToolReleased = function () {
+    this.updateDirtyStatus();
   };
 
-  ns.SavedStatusService.prototype.onToolReleased = function () {
-    this.updateDirtyStatus(true);
+  ns.SavedStatusService.prototype.onPiskelReset = function () {
```

Comment:
Same, let's remove this method and bind PISKEL_RESET to updateDirtyStatus directly